### ISSUE TITLE
Feature: squiggle Sudoku

### ISSUE THREAD

**Issue by andimeier** [andimeier/sudokusolver#7]
Implement squiggle sudoku. Checkout the changes in the merged branch "x-sudoku" to see what had to be added for another Sudoku type.

**Comment by andimeier** [comment]
Also to be solved: how can the squiggles be defined? Should be in the Sudoku input file, e.g. a matrix of "squiggle/box numbers", e.g.:

```
111222333
112233333
111222236
144455566
445556667
444455567
888999667
888999777
888999777
```

This block of squiggles definition should be introduced with the control line "Boxes:" OSLT. So, the Sudoku input file could look like that:

```
type: squiggle
100730093
405007060
405560570
700041501
000000139
044500123
405007060
405560570
700041501
boxes:
111222333
112233333
111222236
144455566
445556667
444455567
888999667
888999777
888999777
```

**Comment by andimeier** [comment]
Jigsaw Sudoku solved with commit 8776148
